src/common/etcd_delegate.rs

```diff
@@ -111,12 +116,96 @@ impl EtcdDelegate {
         })?;
 
         let kvs = resp.take_kvs();
+
         match kvs.get(0) {
-            Some(kv) => Ok(Some(util::decode_from_bytes::<T>(kv.value())?)),
+            Some(kv) => Ok(Some((
+                util::decode_from_bytes::<T>(kv.value())?,
+                kv.version(),
+            ))),
             None => Ok(None),
         }
     }
 
+    /// offer an version check
+    async fn write_to_etcd_with_version<
+        T: DeserializeOwned + Serialize + Clone + Debug + Send + Sync,
+        K: Into<Vec<u8>> + Debug + Clone + Send,
+    >(
+        &self,
+        key: K,
+        value: &T,
+        prev_version: KVVersion,
+        timeout: Option<Duration>,
+    ) -> DatenLordResult<Option<(T, KVVersion)>> {
+        let bin_value = bincode::serialize(value)
+            .with_context(|| format!("failed to encode {value:?} to binary"))?;
+        let mut put_request = etcd_client::EtcdPutRequest::new(key.clone(), bin_value);
+        if let Some(dur) = timeout {
+            put_request.set_lease(
+                self.etcd_rs_client
+                    .lease()
+                    .grant(EtcdLeaseGrantRequest::new(dur))
+                    .await
+                    .with_context(|| {
+                        format!(
+                            "failed to get LeaseGrantResponse from etcd, the timeout={}",
+                            dur.as_secs()
+                        )
+                    })?
+                    .id(),
+            );
+        };
+        let txn_req = etcd_client::EtcdTxnRequest::new()
+            // etcd：chack key exist in txn
+            //  https://github.com/etcd-io/etcd/issues/7115
```

Comment:
It would be better to include this suggestion in the official documentation (https://etcd.io/docs/v3.4/learning/api/).

Reply:
What suggestion?